tests/page-objects/default/about/about.wdio.page.js

```diff
@@ -1,5 +1,6 @@
-const userName = () => $('dt=User name');
+const userName = () => $('label=User name');
 const partners = () => $('.partners');
+const version = () => $('.material .cell p');
```

Comment:
I tend to use IDs only as a last resort, for two reasons. Firstly, it's adding production code purely for testing which isn't ideal. Secondly, this is actually testing the user facing text - the user doesn't care what the ID is but if we change the label to something else we want the test to fail because we've broken the user experience.

Why do you prefer adding IDs?

Reply:
We started using this since the protractor times as a way to identify content, especially dynamic content like contact summaries; otherwise, it would be hard for the e2e to find the correct piece of information to verify correctness. We need to provide enough identifiable elements to facilitate the e2e. 

In this case, the About page isn't dynamic because the labels will always be in the same position, but breaking the e2e test because the element changed position or type is rarely helpful, and what matters the most is that the data is displayed. 

This also comes in handy when the selectors are too complex or inefficient (selectors can also impact performance). Something like this makes it cleaner and faster for those with less experience.  

The `test-id` is harmless and ignored by the browser and processors, so adding it or not, doesn't make a difference for the end user in production. 

In this case, checking that the data is there is the important piece, not the position, so I suggested this to avoid breaking the e2e because of a change in the DOM.

Reply:
> We started using this since the protractor times as a way to identify content,

I agree that with dynamic content it can be difficult to find the exact element you want. This is the last resort where I would use IDs. But as you say, doesn't apply in this case.

> (selectors can also impact performance). 

Not a concern in e2e testing. We have many places where we literally pause the run for seconds at a time so a microsecond of selector performance won't make a difference.

> The test-id is harmless and ignored by the browser and processors

It adds clutter to the template which makes the code harder to maintain.

> In this case, checking that the data is there is the important piece, not the position

I disagree. If the data shows up with a different label then the data is wrong, for example, if we started showing "latin-panda" underneath "supported browser" then that should result in a failing test. With the ID approach it'll pass.

Reply:
>  If the data shows up with a different label then the data is wrong

Then we need to add tests to the labels to know if they change places by mistake 😬 it's a data pair, and it seems reasonable to verify them.

IMHO, the gains are higher than the consequences of adding `test-id`, mainly because e2e tests have become an essential piece in our development experience. But we add them only in specific scenarios where selectors are complex or content is too dynamic. 

I'm not strongly opinionated for the About page, and it's okay to drop the suggestion.